### PR TITLE
Feature/support certificate auth

### LINKED ISSUE #235: [cetic/nifi] client certificate only authentication fails
**Describe the bug**
Client Certificate authentication (only) does not work as expected.

I deployed from the master branch with the following settings:

```
helm install  nifi . --namespace <my namespace> -f values.yaml \
--set registry.enabled=false \
--set auth.singleUser.username='' \
--set auth.singleUser.password='' \
--set properties.needClientAuth='true' \
--set properties.safetyValve."nifi\.security\.user\.login\.identity\.provider"='' \
--set properties.safetyValve."nifi\.security\.user\.authorizer"='managed-authorizer'
```

The deployment fails to start.  Snip of the stack trace:

```
2022-02-22 18:29:11,483 INFO [main] o.a.nifi.properties.NiFiPropertiesLoader Loaded 135 properties from /opt/nifi/nifi-current/./conf/nifi.properties 
2022-02-22 18:29:16,152 ERROR [main] o.s.web.context.ContextLoader Context initialization failed 
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.springframework.security.config.annotation.web.configuration.WebSecurityConfiguration': Unsatisfied dependency expressed through method 'setFilterChainProxySecurityConfigurer' parameter 1; nested exception is org.springframework.beans.factory.BeanExpressionException: Expression parsing failed; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.apache.nifi.web.NiFiWebApiSecurityConfiguration': Unsatisfied dependency expressed through method 'setJwtAuthenticationProvider' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jwtAuthenticationProvider' defined in class path resource [nifi-web-security-context.xml]: Cannot resolve reference to bean 'authorizer' while setting constructor argument; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'authorizer': FactoryBean threw exception on object creation; nested exception is java.lang.reflect.InvocationTargetException 
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.resolveMethodArguments(AutowiredAnnotationBeanPostProcessor.java:768) 
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:720) 
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:119) 
...
```

The statefulset.yaml file needs another block to allow client certificate only authentication.  This change enabled the server to start:

```
{{- else if .Values.auth.singleUser.username }}
          bin/nifi.sh set-single-user-credentials {{ .Values.auth.singleUser.username }} {{ .Values.auth.singleUser.password }}
{{- else }}
          cat "${NIFI_HOME}/conf/authorizers.temp" > "${NIFI_HOME}/conf/authorizers.xml"
{{- end }}
```

Overall, it seems that it would be easier for end-users to indicate client certificate authentication in the auth section of values.yaml.  It would also serve as a reminder to test that execution path of the helm install.  I can work on a PR if needed.

**Version of Helm, Kubernetes and the Nifi chart**:
Helm: 3.5.1
Kubernetes: 1.20.5
Nifi chart: 1.0.5

**What happened**:
Nifi server failed to start.

**What you expected to happen**:
Nifi server starts and supports client certificate authentication.

**How to reproduce it** (as minimally and precisely as possible):
See above.

### PR DESCRIPTION
**Pull request by gforeman02**
#### What this PR does / why we need it:
Addresses Client Certificate authentication limited availability in existing chart.  This is a standard authentication mechanism supported by Nifi. [ref](https://nifi.apache.org/docs/nifi-docs/html/walkthroughs.html#securing-nifi-with-provided-certificates)

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #235

#### Special notes for your reviewer:
The chart version number assumes other PR's I submitted will be approved.  I can adjust as needed.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: doc/USERMANAGEMENT.md [hunk 1]
```diff
@@ -1,24 +1,100 @@
 User Authentication
 =============
 
-This helm chart provides three types of authentication: Single User, LDAP and OIDC. These three authtications can be managed essentialy from the `values.yaml` file. 
+This helm chart provides four types of authentication: Single User, Client Certificate, LDAP, and OIDC. These four authtications can be managed essentialy from the `values.yaml` file.
```
Comment (banzo) [inline doc/USERMANAGEMENT.md:4]: typo in "authtications"

Comment (gforeman02) [reply doc/USERMANAGEMENT.md]: @banzo thank you, corrected.

**Comment by wknickless** [comment]
@gforeman02 have you tried something like https://tanzu.vmware.com/developer/guides/nginx-ingress-certificate-forwarding/ to allow NiFi to see the client certificate when the browser goes through an NGINX proxy?  (https://stackoverflow.com/questions/54617945/client-certificate-authentication-nginx-ssl-pass-through may also be a step in that direction.) 

I haven’t fully wrapped my brain around how that would work, but perhaps it’s a path towards both letting the NGINX proxy provide a certificate to the browser that the browser can trust, and having NGINX pass the client certificate to NiFi for end-to-end browser-to-NiFi authentication.

Writing a test for client certificate authentication ought to be a lot easier than what we did for OIDC testing; no need for Puppeteer and a full-up browser because there’s no browser redirection in the authentication flow.

**Review by zakaria2905 — APPROVED** [review]
We will merge this PR

Thanks for your work

**Comment by gforeman02** [comment]
@zakaria2905 thank you for the review and you are very welcome

**Review by gforeman02 — COMMENTED** [review]
@banzo is it possible to include this in the 1.0.5 release?

**Comment by gforeman02** [comment]
@banzo disregard my question, I did not see the chart progression here: https://artifacthub.io/packages/helm/cetic/nifi/1.0.6

**Review by banzo — APPROVED** [review]

**Pull request merged**